File: codegen/codegen-erlang/src/test/resources/ir/presigner_module.expected.erl
%% Generated presigned URL helper for smithy.beam.test.sigv4#Sigv4TestService.
-module(sigv4test_service_presigner).
-include("runtime_types.hrl").
-export([presign_url/3]).
-type client_config() :: #{binary() => term()}.

-spec presign_url(client_config(), Operation :: atom(), http_request()) ->
    {ok, binary()} | {error, term()}.
presign_url(Config, Operation, Request) ->
    Credentials = maps:get(credentials, Config),
    Region = maps:get(region, Config, <<"us-east-1">>),
    Service = maps:get(signing_name, Config),
    Expires = maps:get(presign_expires, Config, 900),
    Unsigned = maps:get({unsigned_payload, Operation}, Config, false),
    Opts = #{
        expires => Expires,
        unsigned_payload => Unsigned,
        endpoint_host => sigv4test_service_sigv4:endpoint_host_from_config(Config)
    },
    sigv4test_service_sigv4:presign(Request, Credentials, Region, Service, Opts).

File: codegen/codegen-elixir/src/test/resources/ir/presigner_module.expected.ex
defmodule Sigv4testServicePresigner do
  @moduledoc "false"
  alias RuntimeTypes, as: RuntimeTypes
  alias Sigv4testServiceSigv4, as: ServiceSigv4

  @spec presign_url(map(), atom(), RuntimeTypes.HttpRequest.t()) ::
          {:ok, String.t()} | {:error, term()}
  def presign_url(config, operation, request) do
    credentials = Map.fetch!(config, :credentials)
    region = Map.get(config, :region, "us-east-1")
    service = Map.fetch!(config, :signing_name)
    expires = Map.get(config, :presign_expires, 900)
    unsigned = Map.get(config, {:unsigned_payload, operation}, false)
    opts =
      %{
        expires: expires,
        unsigned_payload: unsigned,
        endpoint_host: ServiceSigv4.endpoint_host_from_config(config)
      }
    ServiceSigv4.presign(request, credentials, region, service, opts)
  end
end